mixer/test/client/istio_authn_origin_reject_no_jwt/authn_origin_reject_no_jwt_test.go

```diff
@@ -0,0 +1,74 @@
+// Copyright 2018 Istio Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package istioAuthnOriginRejectNoJwt
+
+import (
+	"fmt"
+	"testing"
+
+	"istio.io/istio/mixer/test/client/env"
+)
+
+// The Istio authn envoy config
+const authnConfig = `
+{
+  "type": "decoder",
+  "name": "istio_authn",
+  "config": {
+    "policy": {
+      "origins": [
+        {
+          "jwt": {
+            "issuer": "issuer@foo.com",
+            "jwks_uri": "http://localhost:8081/"
+          }
+        }
+      ],
+      "principal_binding": 1
+    },
+    "jwt_output_payload_locations": {
+      "issuer@foo.com": "sec-istio-auth-jwt-output"
+    }
+  }
+},
+`
+
+func TestAuthnOriginRejectNoJwt(t *testing.T) {
+	s := env.NewTestSetup(env.IstioAuthnTestOriginRejectNoJwt, t)
+	// In the Envoy config, requires a JWT for origin
+	s.SetFiltersBeforeMixer(authnConfig)
+
+	env.SetStatsUpdateInterval(s.MfConfig(), 1)
+	if err := s.SetUp(); err != nil {
+		t.Fatalf("Failed to setup test: %v", err)
+	}
+	defer s.TearDown()
+
+	url := fmt.Sprintf("http://localhost:%d/echo", s.Ports().ClientProxyPort)
+
+	// Issues a GET echo request with 0 size body
+	tag := "OKGet"
+
+	// No jwt_auth header to be consumed by Istio authn filter.
+	// The request will be rejected by Istio authn filter.
+	code, _, err := env.HTTPGet(url)
+	if err != nil {
+		t.Errorf("Failed in request %s: %v", tag, err)
+	}
+	// Verify that the http request is rejected
+	if code != 401 {
```

Comment:
do we have error message in the response body?  if we do, could we verify it too?

Reply:
Done: added the verification of the error messages.